fixed quoting

machisuji opened the pull request:
Element IDs were out of sync. Added feature spec.

WP [#19549](https://community.openproject.org/work_packages/19549)

machisuji committed:
fixed quoting

Element IDs were out of sync. Added feature spec.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://machisuji-fix-forum-quote-message.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://machisuji-fix-forum-quote-message.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

myabc commented on spec/features/boards/message_spec.rb:
```diff
@@ -0,0 +1,70 @@
+        within(msg) { click_on 'Quote' }
+
+        reply = find '#reply'
+        expect(reply.visible?).to be true
```
@machisuji 

``` ruby
expect(reply).to be_visible
```

reads better IMO

NobodysNightmare commented:
@machisuji Specs are failing, looks like legitimate failures...

machisuji commented:
Yeah ...

```
 3) Error:
ActivitiesControllerTest#test_previous_project_index:
ActionView::Template::Error: couldn't find file 'bundles/openproject-global'
```

machisuji commented:
There are legitimate ones too, though. Yes. I'm looking into it.

NobodysNightmare commented on app/controllers/messages_controller.rb:
```diff
@@ -146,12 +146,12 @@ def quote
-      page << "$('message_subject').value = \"#{subject}\";"
-      page.<< "$('message_content').value = \"#{content}\";"
+      page << "$('reply_subject').value = \"#{subject}\";"
+      page.<< "$('reply_content').value = \"#{content}\";"
```
That `$` in this ruby file, is this the jQuery `$`?

Urgh!

myabc replied:
@NobodysNightmare no, that's Prototype.

NobodysNightmare replied:
Doesn't make it better ;)

NobodysNightmare commented on app/controllers/messages_controller.rb:
```diff
@@ -146,12 +146,12 @@ def quote
-      page.<< "$('message_content').value = \"#{content}\";"
+      page << "$('reply_subject').value = \"#{subject}\";"
+      page.<< "$('reply_content').value = \"#{content}\";"
       page.show 'reply'
```
Also looking totally legit to have that `page.show` between the `page << foo` lines... of course... mixing `page.<<` and `page <<` which is probably the same...

Not that any of this is your fault, but this looks like a big pile-o-WTF...

myabc replied:
I'd suggest we use every opportunity we can get to rip out Prototype, even if out-of-scope (arguably). See fda43ac3045dd18cadcfb331e266373f572fc742 as an example.

machisuji replied:
The good thing is, once the test is green we can safely refactor this thing of beauty.
Thanks for the pointer @myabc. I shall have a go at it.

myabc replied:
@machisuji I agree. Fixing and _then_ refactoring is optimal.

machisuji replied:
Too bad the feature spec fails with mysql2, though not with postgres... Wouldn't have thought that the db matters.

NobodysNightmare replied:
@machisuji I think that problem is solved with https://travis-ci.org/opf/openproject/builds/59219347 failing across the board ^^

machisuji replied:
That's actually because of another issue I tried to fix. But I reverted the change for now.

The problem is, that the forum message text area has the ID `message_content` when you create a topic and `reply_content` when you reply to a topic or post. The [wiki toolbar](https://github.com/opf/openproject/blob/dev/app/views/messages/_form.html.erb#L49) is only added for the ID `message_content`. Which means you will have it when you create a topic, but not when you answer to one.

I'm thinking instead of changing the controller code to `reply_*` I should probably change the markup back to `message_*` even for replies. Although I'm not sure if there was a more important reason to rename it to begin with.

NobodysNightmare commented:
Added the WIP label, you are still working on that one, aren't you?

machisuji commented:
True. The specs are finally green, yay. So I can do a bit of refactoring now, throwing out Prototype, and then I'm done.

myabc commented:
@machisuji can you squash/rebase out cd256f9?

machisuji committed:
fixed wiki toolbar

Use reply or message content depending on the context.
Perhaps one should just always call it message instead of reply
but I can't say what the full impact of that would be.

So in doubt this is the simpler solution.

machisuji committed:
jQuery all the way

machisuji commented:
Please note that the failing specs are unrelated to this PR. The spec for this PR itself and the other board related specs are green.

NobodysNightmare commented:
I don't believe you... Are you telling me, that the specs are already failing on the ~~`dev`~~ `release/4.1` branch?

Otherwise there is no way, that you did not break it ^^

machisuji commented:
That was referring to the state before d263823. Now d263823 does fail, but that's because of the refactoring and a test (not spec) that tests against the old implementation. I'll adjust the test. Then everything should be green.

machisuji committed:
adjusted test after refactoring

NobodysNightmare commented:
LGTM (if travis agrees) :+1:

myabc commented on app/views/messages/quote.js.erb:
```diff
@@ -0,0 +1,35 @@
+<%#-- copyright
```
@machisuji the copyright header is missing some line breaks.

myabc commented on app/views/messages/quote.js.erb:
```diff
@@ -0,0 +1,35 @@
+reply.slideDown();
+content.focus();
+
+jQuery('html, body').animate({
```
:+1:

floriank replied:
that's just me being a style nazi and even this is a response from a controller which will be properly wrapped on call, I'd prefer wraping it in an extra closure using `$` for `jQuery` instead

``` javascript
(function($) {
  var reply = $('#reply'),
      subject = $('#reply_subject'),
      // [..]
}(jQuery))
```

myabc replied:
yep, agreed.

myabc commented:
LGTM too!

machisuji committed:
$$$ = sexy

machisuji committed:
added missing newlines

floriank commented:
LGTM, merge it when travis says ok.

NobodysNightmare commented:
Great, you've accumulated 3x LGTM!

:+1: :+1: :+1:

floriank commented:
now go and get me 20 rugged hides!

NobodysNightmare commented:
@Azure7111 what do you think?

machisuji commented:
I'm expecting a purple reward.

Azure7111 commented:
Your reading skill has increased by one point.

floriank commented:
Purple? Here's a grey and 50 copper.

machisuji commented:
This leaves me [disappointed](https://www.youtube.com/watch?v=_O1hM-k3aUY&t=10).
